src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -128,68 +128,96 @@ private struct Inline3ByteArray
         public int CurrentDepth => _currentDepth & JsonConstants.RemoveFlagsBitMask;
 
         /// <summary>
-        /// Length of the partial code point.
+        /// Length of the partial string data.
         /// </summary>
-        private byte PartialCodePointLength
+        private byte PartialStringDataLength
         {
-            get => (byte)(_partialCodePointFlags & PartialCodePointLengthMask);
-            set => _partialCodePointFlags = (byte)((_partialCodePointFlags & ~PartialCodePointLengthMask) | (byte)value);
+            get => (byte)(_partialStringDataFlags & PartialStringDataLengthMask);
+            set => _partialStringDataFlags = (byte)((_partialStringDataFlags & ~PartialStringDataLengthMask) | value);
         }
 
         /// <summary>
         /// The partial UTF-8 code point.
         /// </summary>
-        private ReadOnlySpan<byte> PartialUtf8CodePoint
+        private ReadOnlySpan<byte> PartialUtf8StringData
         {
             get
             {
                 Debug.Assert(PreviousSegmentEncoding == SegmentEncoding.Utf8);
```

Comment:
I'm guessing `PreviousSegmentEncoding` would eventually be replaced by "enclosing container" data as introduced in #111332

Reply:
That PR doesn't do it but it could in order to avoid the bit twiddling that `PreviousSegmentEncoding` currently does.